### PR TITLE
Add auto deployment for GitHub Pages

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: .github/workflows/ghpage.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: GitHub Pages
+
+on:
+  push:
+    branches:
+      - mainline  # Set a branch name to trigger deployment
```
Comment (zxkane) [inline .github/workflows/ghpage.yml]: use `develop` branch for preview

File: .github/workflows/ghpage.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0    # Fetch all history for .GitInfo and .Lastmod
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.15.1'
```
Comment (zxkane) [inline .github/workflows/ghpage.yml]: why pin to a specific version?

**Review by zxkane — CHANGES_REQUESTED** [review]
one more question,

is it tested in your forked repo? any preview link?

File: .github/workflows/ghpage.yml (hunk 1 above)
Comment (ghost) [reply .github/workflows/ghpage.yml]: done [8d618f9](https://github.com/awslabs/service-workbench-on-aws-cn/pull/21/commits/8d618f96682f3d2f5812f8e51117cd9d5955c90c)

File: .github/workflows/ghpage.yml (hunk 2 above)
Comment (ghost) [reply .github/workflows/ghpage.yml]: removed [7c1aa4d](https://github.com/awslabs/service-workbench-on-aws-cn/pull/21/commits/7c1aa4dada04fa5cc799dcb7ee95e32f06dadab1)

**Review by zxkane — APPROVED** [review]

File: .github/workflows/ghpage.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        working-directory: ./docs
+        run: |
+          npm install 
+      
+      - name: Build docs
+        working-directory: ./docs
+        run: |
+          yarn build
+
+      - name: Deploy
```
Comment (zxkane) [inline .github/workflows/ghpage.yml:30]: it should only happen when the release branch is triggered not all pull request checking runs

**Review by zxkane — CHANGES_REQUESTED** [review]

File: .github/workflows/ghpage.yml (hunk 3 above)
Comment (ghost) [reply .github/workflows/ghpage.yml:30]: relevant constraints added

**Comment by ghost** [comment]
> one more question,
> 
> is it tested in your forked repo? any preview link?

Yes and it can be viewed on the [github.io pages](https://6zkf6zs05ryz.github.io/service-workbench-on-aws-cn/) of my repo

**Review by zxkane — APPROVED** [review]

**Pull request merged**